packages/multichain/src/scope/authorization.ts

```diff
@@ -51,3 +52,42 @@ export const validateAndNormalizeScopes = (
     normalizedOptionalScopes,
   };
 };
+
+/**
+ * Groups a NormalizedScopesObject into three separate
+ * NormalizedScopesObjects for supported scopes,
+ * supportable scopes, and unsupportable scopes.
+ * @param scopes - The NormalizedScopesObject to group.
+ * @param hooks - The hooks.
+ * @param hooks.isChainIdSupported - A helper that returns true if an eth chainId is currently supported by the wallet.
+ * @param hooks.isChainIdSupportable - A helper that returns true if an eth chainId could be supported by the wallet.
+ * @returns an object with three NormalizedScopesObjects separated by support.
+ */
+export const bucketScopes = (
+  scopes: NormalizedScopesObject,
+  {
+    isChainIdSupported,
+    isChainIdSupportable,
+  }: {
+    isChainIdSupported: (chainId: Hex) => boolean;
```

Comment:
What's the plan for making this non-EVM compatible?

Reply:
I think this will merit some conversation. We will possibly/probably want to figure out the right strategy for giving dapps the same capability of suggesting the user add EVM networks for non EVM networks (by handing us the snap information as part of the CAIP-25 request). But since this feature itself is not part of V1 I think its okay to leave it as TBD for now?